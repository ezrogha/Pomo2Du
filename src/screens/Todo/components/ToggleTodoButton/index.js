import React from 'react'
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';

export default ({ todoTabState, doneTabState, changeTab }) => {
    console.log(`Todo: ${todoTabState}, Done: ${doneTabState}`)
    return (
    <View style={styles.view}>
        <TouchableOpacity activeOpacity={1} onPress={() => changeTab(0)} style={[ styles.button, todoTabState && styles.active ]}>
            <Text style={[ styles.text, todoTabState &&  styles.activeText ]}>Todo</Text>
        </TouchableOpacity>
        <TouchableOpacity activeOpacity={1} onPress={() => changeTab(1)} style={[ styles.button, doneTabState && styles.active ]}>
            <Text style={[ styles.text, doneTabState &&  styles.activeText ]}>Done</Text>
        </TouchableOpacity>
    </View>
)}

const styles = StyleSheet.create({
    view: {
        flexDirection: "row",
        borderWidth: 1,
        borderColor: "white",
        borderRadius: 2
    },
    button: {
        paddingTop: 5,
        paddingBottom: 5,
        paddingLeft: 10,
        paddingRight: 10,
    },
    active: {
        backgroundColor: "white"
    },
    activeText: {
        color: "#009DDD"
    },
    default: {
        backgroundColor: "#009DDD"
    },
    defaultText: {
        color: "white"
    },
    text: {
        color: "white"
    }
})