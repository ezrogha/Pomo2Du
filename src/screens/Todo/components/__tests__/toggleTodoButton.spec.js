import React from 'react'

import ToggleTodoButton from '../ToggleTodoButton';
import renderer from 'react-test-renderer'

describe("Testing ToggleTodoButton Component", () => {
    test("snapshot test", () => {
        const props = {
            todoTabState: false,
            doneTabState: true,
            changeTab: () => {},
        }
        const tree = renderer.create(<ToggleTodoButton { ...props } />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
