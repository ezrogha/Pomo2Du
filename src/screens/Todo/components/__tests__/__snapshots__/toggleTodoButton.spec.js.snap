// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ToggleTodoButton Component snapshot test 1`] = `
<View
  style={
    Object {
      "borderColor": "white",
      "borderRadius": 2,
      "borderWidth": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
          false,
        ]
      }
    >
      Todo
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "opacity": 1,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
          Object {
            "color": "#009DDD",
          },
        ]
      }
    >
      Done
    </Text>
  </View>
</View>
`;
